### ISSUE TITLE
AsdfInFits deprecation and move to stdatamodels

### ISSUE THREAD

**Issue by braingram** [astropy/specutils#1006]
- [ ] Also revert https://github.com/astropy/specutils/pull/1021 if applicable
- [ ] Also revert https://github.com/astropy/specutils/pull/1024 if applicable

Blocked by:

*  spacetelescope/stdatamodels#134

<hr/>

The ability to read AsdfInFits files is being moved from [ASDF](https://github.com/asdf-format/asdf) into [stdatamodels](https://github.com/spacetelescope/stdatamodels) (in part to consolidate non-pipeline code for jwst). The current plan for this move is:

- include AsdfInFits in stdatamodels: https://github.com/spacetelescope/stdatamodels/pull/110
- ~~release a stable version of stdatamodels~~ EDIT: see comment below on stdatamodels >1.0
- drop AsdfInFits support in ASDF 3.0: https://github.com/asdf-format/asdf/pull/1288

We do not have a estimate for when a stable version of stdatamodels or ASDF 3.0 will be released. We are planning to release ASDF 2.15 (which will support AsdfInFits) in the next few weeks which we will continue to support for at least 1 year (please see our proposed [release and support process](https://github.com/asdf-format/asdf/pull/1323)).

To summarize, at some point in the next year or so the uses of AsdfInFits in specutils and opening of fits files using `asdf.open` will need to be updated.

I've found a usage of AsdfInFits in a test fixture
https://github.com/astropy/specutils/blob/fc04cd6c948698c70ec9bd15b3354b58e4dbbcc2/specutils/io/default_loaders/tests/test_jwst_reader.py#L407-L411
and likely usage in jwst_reader.py
https://github.com/astropy/specutils/blob/fc04cd6c948698c70ec9bd15b3354b58e4dbbcc2/specutils/io/default_loaders/jwst_reader.py#L557-L559

I am happy to discuss alternatives, suggestions, comments on this process and hope to make it as painless as possible.

Also see:

* #1011

[🐱](https://jira.stsci.edu/browse/JDAT-3160)

**Comment by pllim** [comment]
Will the work to address this also naturally address #1011 ?

**Comment by braingram** [comment]
The two issues are mostly unrelated (aside from both relying on deprecated or soon to be deprecated ASDF features).

To provide a minor update to this issue, a version of stdatamodels with AsdfInFits support has been released:
https://pypi.org/project/stdatamodels/1.1.0/
with documentation on reading and writing AsdfInFits files: https://stdatamodels.readthedocs.io/en/latest/asdf_in_fits.html
